lib/utils/utils.js

```diff
@@ -0,0 +1,36 @@
+const prefix = (source) => {
+  switch (source) {
+    case 'sierra-nypl':
+      return 'b'
+    case 'recap-pul':
+      return 'pb'
+    case 'recap-cul':
+      return 'cb'
+    case 'recap-hl':
+      return 'hb'
+    default:
+      return ''
+  }
```

Comment:
We should really have this be driven from [the central "source mapping" file in NYPL-core](https://github.com/NYPL/nypl-core/blob/master/mappings/recap-discovery/nypl-source-mapping.json). There's [a util we have used](https://github.com/NYPL-discovery/discovery-store-models/blob/40acac46a231b9eac185e7670df7eea5480378c6/lib/nypl-source-mapper.js) for interfacing that data when needing to convert prefixed identifiers into their component keys (id, nyplSource). You want to do the reverse, but there will likely be a need for converting in both directions, so I'd copy that NyplSourceMapper lib into this repo and amend it to your needs. (Note that the `syncRequest` call isn't going to work in Node 18, so you'll need to make it async.) What you need to do here is convert an nyplSource into a prefix, which could be done by loading the [above source mapping json](https://github.com/NYPL/nypl-core/blob/master/mappings/recap-discovery/nypl-source-mapping.json) and then returning `mapping[nyplSource].bibPrefix`. We don't add partners often, but it sure helps to have it centralized